### PR TITLE
feat(swb): Merge BYON into develop

### PR DESCRIPTION
**Pull request by kpark277**
Issue #, if available:

Description of changes:
* Merging feat/swb-byon branch into develop
* Switches UI to use ECS and ALB

Previous PRs this merge consisted of:
* https://github.com/aws-solutions/solution-spark-on-aws/pull/722
* https://github.com/aws-solutions/solution-spark-on-aws/pull/708
* https://github.com/aws-solutions/solution-spark-on-aws/pull/683
* https://github.com/aws-solutions/solution-spark-on-aws/pull/678
* https://github.com/aws-solutions/solution-spark-on-aws/pull/669
* https://github.com/aws-solutions/solution-spark-on-aws/pull/594
* https://github.com/aws-solutions/solution-spark-on-aws/pull/586
* https://github.com/aws-solutions/solution-spark-on-aws/pull/583
* https://github.com/aws-solutions/solution-spark-on-aws/pull/570
* https://github.com/aws-solutions/solution-spark-on-aws/pull/565
* https://github.com/aws-solutions/solution-spark-on-aws/pull/555
* https://github.com/aws-solutions/solution-spark-on-aws/pull/545
* https://github.com/aws-solutions/solution-spark-on-aws/pull/535
* https://github.com/aws-solutions/solution-spark-on-aws/pull/526

Checklist:

* [x] Have you successfully deployed to an AWS account with your changes?
* [x] Have you written new tests for your core changes, as applicable?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by DaveMadsen — APPROVED** [review]

File: solutions/swb-ui/ui/package.json [hunk 1]
```diff
@@ -21,7 +21,7 @@
     "build:test": "rushx build && rushx test",
     "check-license-header": "license-check-and-add check -f license-add-config.json",
     "depcheck": "depcheck",
-    "deploy-ui-and-api": "bash ./scripts/deploy.sh",
+    "deploy-ui-and-api": "bash ./scripts/image-deploy.sh $BRANCH && bash ./scripts/deploy.sh",
```
Comment (SanketD92) [inline solutions/swb-ui/ui/package.json:24]: How is `$BRANCH` resolved here, and could we pin it to a specific branch (like develop) till the release?

Comment (SanketD92) [reply solutions/swb-ui/ui/package.json:24]: nvm, I see it's set to develop if not passed

**Review by SanketD92 — CHANGES_REQUESTED** [review]
LGTM

**Review by SanketD92 — APPROVED** [review]
LGTM

File: solutions/swb-ui/ui/package.json (hunk 1 above)
Comment (kpark277) [reply solutions/swb-ui/ui/package.json:24]: `$BRANCH` gets passed in as an env var like STAGE; if none is passed in, the image-deploy.sh script will default to use develop.

**Pull request merged**